Source/queries/Query.ts

```diff
@@ -1,13 +1,29 @@
 /*---------------------------------------------------------------------------------------------
- *  Copyright (c) Dolittle. All rights reserved.
- *  Licensed under the MIT License. See LICENSE in the project root for license information.
- *--------------------------------------------------------------------------------------------*/
+*  Copyright (c) Dolittle. All rights reserved.
+*  Licensed under the MIT License. See LICENSE in the project root for license information.
+*--------------------------------------------------------------------------------------------*/
+import { IQuery } from "./internal";
 
 /**
  * Defines the base of a query
+ *
+ * @export
+ * @class Query
+ * @implements {IQuery}
  */
- export class Query {
-     nameOfQuery: string="";
-     generatedFrom: string="";
-     readModel=null;
-}
\ No newline at end of file
+export abstract class Query implements IQuery {
```

Comment:
Same comment as before on abstract classes - only use it when it is abstract

Reply:
Same as my comment above :)